src/prefect/environments/storage/gitlab.py

```diff
@@ -0,0 +1,161 @@
+    ```
+
+    - Push this `flow.py` file to the `my/repo` repository under `/flows/flow.py`.
+
+    - Call `prefect register -f flow.py` to register this flow with GitLab storage.
+
+    Args:
+        - repo (str): the project path (i.e., '<namespace>/<project>') or ID
+        - host (str, optional): If using Gitlab server, the server host. If not specified, defaults
+            to Gitlab cloud.
+        - path (str, optional): a path pointing to a flow file in the repo
+        - ref (str, optional): a commit SHA-1 value or branch name
+        - **kwargs (Any, optional): any additional `Storage` initialization options
+    """
+
+    def __init__(
+        self,
+        repo: str,
+        host: str = None,
+        path: str = None,
+        ref: str = None,
+        **kwargs: Any,
+    ) -> None:
+        self.flows = dict()  # type: Dict[str, str]
+        self._flows = dict()  # type: Dict[str, "Flow"]
+        self.repo = repo
+        self.host = host
+        self.path = path
+        self.ref = ref
+
+        super().__init__(**kwargs)
+
+    @property
+    def default_labels(self) -> List[str]:
+        return ["gitlab-flow-storage"]
+
+    def get_flow(self, flow_location: str = None, ref: str = None) -> "Flow":
+        """
+        Given a flow_location within this Storage object, returns the underlying Flow (if possible).
+        If the Flow is not found an error will be logged and `None` will be returned.
+
+        Args:
+            - flow_location (str): the location of a flow within this Storage; in this case,
+                a file path on a repository where a Flow file has been committed. Will use `path` if not
+                provided.
+            - ref (str, optional): a commit SHA-1 value or branch name. Defaults to 'master' if
+                not specified
+
+        Returns:
+            - Flow: the requested Flow
+
+        Raises:
+            - ValueError: if the flow is not contained in this storage
+            - UnknownObjectException: if the flow file is unable to be retrieved
+        """
+        if flow_location:
+            if flow_location not in self.flows.values():
+                raise ValueError("Flow is not contained in this Storage")
+        elif self.path:
+            flow_location = self.path
+        else:
+            raise ValueError("No flow location provided")
+
+        # Use ref argument if exists, else use attribute, else default to 'master'
+        ref = ref if ref else (self.ref if self.ref else "master")
+
+        from gitlab.exceptions import GitlabAuthenticationError, GitlabGetError
+
+        try:
+            project = self._gitlab_client.projects.get(quote_plus(self.repo))
+            contents = project.files.get(file_path=flow_location, ref=ref)
+        except GitlabAuthenticationError as exc:
+            self.logger.error(
+                "Unable to authenticate Gitlab account. Please check your credentials."
+            )
+            raise exc
```

Comment:
When re-raising an exception, you should use `raise` rather than `raise exc`, as the former doesn't add a new chained exception, while the latter does (resulting in a larger traceback).

```suggestion
            raise
```